modules/runner/src/main/java/org/apache/ignite/internal/app/IgnitionImpl.java

```diff
@@ -103,13 +104,16 @@ public CompletableFuture<Ignite> start(
         );
     }
 
-    /** {@inheritDoc} */
     @Override
     public void stop(String nodeName) {
         readyForInitNodes.remove(nodeName);
 
         nodes.computeIfPresent(nodeName, (name, node) -> {
-            node.stop();
+            try {
+                node.stop();
+            } catch (Exception e) {
+                throw new IgniteException(Common.NODE_STOPPING_ERR, e);
```

Comment:
Is this a good exception for the purpose? This is a public API, and according to the error handling design, all public error codes should be associated with the specific implementation class. Ideally, it also should be documented in the method that throws the exception. I don't see any of that.
What was the reason here?

Reply:
I don’t quite understand what you are proposing, should I describe the exception in the documentation? create a new exception? swallow exception from `node.stop()`?

Reply:
What exception is thrown from the `node.stop()`? My proposal is not to throw random `IgniteException` instances in violation of existing error handling rules. Maybe we need a separate issue for the fix, I would accept that.
Are you familiar with the way we treat public exceptions?

Reply:
Why is there suddenly a new exception from the stop method? Is this because of new managers that you close? I propose changing that code instead, and reverting this particular change.

Reply:
> What exception is thrown from the `node.stop()`?

java.lang.Exception

> Are you familiar with the way we treat public exceptions?

No.

> Why is there suddenly a new exception from the stop method?

I think the previous code was wrong
`    public void stop() {
        lifecycleManager.stopNode();
        restAddressReporter.removeReport();
    }`
if we had an error from `stopNode` then we would not execute `removeReport`, which does not seem correct.

Reply:
Fix it